Sources/XcodeGenKit/PBXProjGenerator.swift

```diff
@@ -681,29 +681,34 @@ public class PBXProjGenerator {
         return "\(carthagePath)/\(platformName)"
     }
 
-    func getAllCarthageDependencies(target: Target, visitedTargets: [String: Bool] = [:]) -> [Dependency] {
-
+    func getAllCarthageDependencies(target: Target) -> [Dependency] {
         // this is used to resolve cyclical target dependencies
-        var visitedTargets = visitedTargets
-        visitedTargets[target.name] = true
-
+        var visitedTargets: Set<String> = []
         var frameworks: [Dependency] = []
 
-        for dependency in target.dependencies {
-            switch dependency.type {
-            case .carthage:
-                frameworks.append(dependency)
-            case .target:
-                let targetName = dependency.reference
-                if visitedTargets[targetName] == true {
-                    return []
-                }
-                if let target = project.getTarget(targetName) {
-                    frameworks += getAllCarthageDependencies(target: target, visitedTargets: visitedTargets)
+        var queue: [Target] = [target]
+        while !queue.isEmpty {
+            let target = queue.removeFirst()
+            if visitedTargets.contains(target.name) {
+                continue
+            }
+
+            for dependency in target.dependencies {
+                switch dependency.type {
+                case .carthage:
+                    frameworks.append(dependency)
```

Comment:
Can this result in `frameworks` having duplicate dependencies in this list, and if so should this function return a unique list of dependencies?

Reply:
The carthage dependencies are de-duplicated later on when they are added to a target. This could be done here, but maybe it might be useful at some point to know how many times a dependency is being used